fix(filetype): handle invalid `bufnr` in `_getlines()` and `_getline()`

xvzc opened the pull request:
fix(filetype): handle invalid `bufnr` in `_getlines()` and `_getline()`

This fixes an unexpected behavior where `vim.filetype.match({ filename = 'a.x' })`
returns `nil` even though it should clearly return `"x"`, where the extension `x`
could be one of dozens of possible extensions.

**Problem:**  
`vim.filetype.match({ filename = 'a.sh' })` returns `nil` because an invalid buffer ID
is passed to `vim.api.nvim_buf_get_lines()`. For `csh`, `txt`, or any other extensions
that call `_getlines()` or `_getline()`, the same issue occurs as with `.sh`.

When only the `filename` argument is passed, an error is raised while executing
the filetype-detection function, which is wrapped in a `pcall()`. As a result, it
returns no value and shows no error message.

**Solution:**  
Validate the `bufnr` value in `_getlines()` and `_getline()`.

# Reproduce
```sh
nvim --clean --headless +":lua print(vim.filetype.match({ filename = 'foo.cpp' }))" +q

# OUTPUT: 
# 'cpp nil'
```

```sh
nvim --clean --headless +":lua print(vim.filetype.match({ filename = 'foo.sh' }))" +q

# OUTPUT: 
# ''
```

<!--
  Thank you for contributing to Neovim!
  If this is your first time, check out https://github.com/neovim/neovim/blob/master/CONTRIBUTING.md#pull-requests-prs
  for our PR guidelines.
-->

zeertzjq reviewed (commented):
The problem may still happen for some other extensions (e.g. `.install` files). I wonder if this change should be done in `_getline()` and `_getlines()` instead.

xvzc committed:
fix(filetype): handle invalid `bufnr` in `_getlines()`, `_getline()`

 **Problem:**
`vim.filetype.match({ filename = 'a.sh' })` returns `nil` because
an invalid buffer ID is passed to `vim.api.nvim_buf_get_lines()`.
For filetypes like `csh`, `txt`, or any other extensions that call
`_getlines()` or `_getline()` to detect their filetypes, the same
issue occurs.

When only the `filename` argument is passed, an error is raised
inside a `pcall()` that wraps the filetype detection function,
causing it to return no value without showing any error message.

**Solution:**
Validate the `bufnr` value in `_getlines()` and `_getline()`.

xvzc committed:
test(filetype): add tests for filenames with `.sh`, `.txt`

xvzc committed:
test(fileype): add Pull Request ID

xvzc committed:
refactor(filetype): fix code style
